hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -698,7 +686,59 @@ public String getPayloadClass() {
    * Read the payload class for HoodieRecords from the table properties.
    */
   public String getRecordMergerStrategy() {
-    return getStringOrDefault(RECORD_MERGER_STRATEGY);
+    return getString(RECORD_MERGER_STRATEGY);
+  }
+
+  public static Triple<RecordMergeMode, String, String> inferCorrectMergingBehavior(RecordMergeMode recordMergeMode, String payloadClassName, String recordMergerStrategy) {
+    RecordMergeMode inferRecordMergeMode;
+    String inferPayloadClassName;
+    String inferRecordMergerStrategy;
+
+    if (isNullOrEmpty(payloadClassName)) {
+      if (isNullOrEmpty(recordMergerStrategy)) {
+        checkArgument(recordMergeMode != RecordMergeMode.CUSTOM, "Custom merge mode should only be used if you set a merge strategy");
+        if (recordMergeMode == null) {
+          inferRecordMergeMode = RECORD_MERGE_MODE.defaultValue();
+        } else {
+          inferRecordMergeMode = recordMergeMode;
+        }
+        inferRecordMergerStrategy = HoodieRecordMerger.getAvroMergerStrategyFromMergeMode(inferRecordMergeMode);
```

Comment:
If `HoodieRecordMerger.getAvroMergerStrategyFromMergeMode` is only used here, it's better to embed to switch-case logic here directly for clarity.